### PR TITLE
Cosmetics and small things

### PR DESCRIPTION
**Pull request by cedr-x**
Hello,
Thanks for the project !
I've enhanced the preview cosmetics (less flickering)
please review

### CODE REVIEW

**Comment by cedr-x** [comment]
- output filter option in order to have, for eg, jpg as the output format

File: index.html [hunk 1]
```diff
@@ -123,7 +123,7 @@
                         </thead>
                         <tbody data-bind="foreach: files">
                             <tr>
-                                <td><a data-bind="attr: { href: fullname }, text: name"></a></td>
+                                <td><a data-bind="attr: { href: fullname }, text: name" download></a></td>
```
Comment (cedr-x) [inline index.html:126]: http://www.w3schools.com/tags/att_a_download.asp

**Review by sbs20 — COMMENTED** [review]
Hello! Thank you for your suggestions - I like it all and really appreciate it!

One thing: I'd suggest making api.php:38 `$scanRequest->outputFilter = Config::PreviewFilter;` rather than `"/bin/cat"` and update /classes_php/Config.php accordingly. I think it would be better to have anything which may differ between installs in configuration.

And possibly doing similar things (although this is less important) with line 46 `-trim -quality 30`.

What do you think?

Thanks again,

Sam

**Comment by cedr-x** [comment]
Hello Sam,
I've added the PreviewFilter to api.php, not on  HandlePreviewRequest, but on HandlePreviewToJpegRequest instead. This way, trim and quality are settable in Config.php.
To avoid the use of the dirty "/bin/cat", a test is made in Scanimage if the "outputfilter" is given or not

I've also inserted the jpg output preview within the json output in order to suppress the next http request to preview.jpg

Hope you'll like it !
Regards

**Review by sbs20 — APPROVED** [review]
Awesome - thank you!

**Pull request merged**